### ISSUE TITLE
💅 Import statement with default and named imports – `correctness/noUnusedImports` rule produces invalid JS

### ISSUE THREAD

**Issue by quiqueg** [biomejs/biome#6229]
### Environment information

```bash
CLI:
  Version:                      2.0.0-beta.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm
  JS_RUNTIME_VERSION:           v22.15.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.11.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    correctness/noUnusedImports

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedImports

### Playground link

https://next.biomejs.dev/playground/?enabledAssist=false&code=aQBtAHAAbwByAHQAIABVAHMAZQBkACwAIAB7ACAATgBvAHQAVQBzAGUAZAAgAH0AIABmAHIAbwBtACAAJwBmAG8AbwAnADsACgAKAFUAcwBlAGQAKAApADsA&language=js&ruleDomains.react=recommended&ruleDomains.test=recommended

### Expected result

When the `noUnusedImports` rule is enabled, and an import statement contains both a (used) default import and a (unused) named import, the linter produces invalid code.

Original code to be linted:
```ts
import Used, { NotUsed } from 'foo';

Used();
```

Expected result (valid code):

```ts
import Used from 'foo';

Used();
```

Actual result (invalid code):

```ts
import Usedfrom 'foo';

Used();
```

In the Playground, I can't seem to reproduce with just the `noUnusedImports` selected; but I can reproduce with `recommended` selected.

Locally with the CLI, I can reproduce with just `noUnusedImports` enabled:

```json
{
  "$schema": "https://biomejs.dev/schemas/2.0.0-beta.6/schema.json",
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": false,
      "correctness": {
        "noUnusedImports": "error"
      }
    },
    "includes": [
      "**"
    ]
  }
}
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
